skiko/src/jvmMain/kotlin/org/jetbrains/skiko/Library.kt

```diff
@@ -9,19 +9,39 @@ import java.nio.file.StandardCopyOption
 object Library {
     private val skikoLibraryPath = System.getProperty("skiko.library.path")
     private val cacheRoot = "${System.getProperty("user.home")}/.skiko/"
+    private var copyDir: File? = null
 
-    private fun loadOrGet(cacheDir: File, path: String, resourceName: String, isLibrary: Boolean) {
-        val file = File(cacheDir, resourceName)
+    // Same native library cannot be loaded in several classloaders, so we have to clone
+    // native library to allow Skiko loading to work properly in complex cases, i.e.
+    // several IDEA plugins.
+    private fun loadLibraryOrCopy(library: File) {
+        try {
+            System.load(library.absolutePath)
+        } catch (e: UnsatisfiedLinkError) {
+            if (e.message?.contains("already loaded in another classloader") == true) {
+                val tempFile = File.createTempFile("skiko", if (hostOs.isWindows) "dll" else "")
+                copyDir = tempFile.parentFile
+                Files.copy(library.toPath(), tempFile.toPath(), StandardCopyOption.REPLACE_EXISTING)
+                tempFile.deleteOnExit()
+                System.load(tempFile.absolutePath)
+            } else {
+                throw e
+            }
+        }
+    }
+
+    private fun unpackIfNeeded(dest: File, resourceName: String, deleteOnExit: Boolean): File {
+        val file = File(dest, resourceName)
         if (!file.exists()) {
-            val tempFile = File.createTempFile("skiko", "", cacheDir)
-            Library::class.java.getResourceAsStream("$path$resourceName").use { input ->
+            val tempFile = File.createTempFile("skiko", "", dest)
```

Comment:
shouldn't we use here `resourceName` instead of `skiko`? and also add extension?

Reply:
Not really, we use temp file for unpacking only, and we move it to dest file right after unpack.